src/array_serde.rs

```diff
@@ -139,21 +139,61 @@ impl<A, Di, S> serde::de::Visitor for ArrayVisitor<S,Di>
 {
     type Value = ArrayBase<S, Di>;
 
+    fn visit_seq<V>(&mut self, mut visitor: V) -> Result<ArrayBase<S, Di>, V::Error>
+        where V: serde::de::SeqVisitor
+    {
+        let v: u8 = match try!(visitor.visit()) {
+            Some(value) => value,
+            None => {
+                try!(visitor.end());
+                return Err(serde::de::Error::invalid_length(0));
+            }
+        };
+
+        let dim: Di = match try!(visitor.visit()) {
+            Some(value) => value,
+            None => {
+                try!(visitor.end());
+                return Err(serde::de::Error::invalid_length(1));
+            }
+        };
+
+        let data: Vec<A> = match try!(visitor.visit()) {
+            Some(value) => value,
+            None => {
+                try!(visitor.end());
+                return Err(serde::de::Error::invalid_length(2));
+            }
+        };
+
+        try!(visitor.end());
+
+        if v != ARRAY_FORMAT_VERSION {
```

Comment:
I would check this immediately after deserializing `v`, in case other formats use different fields. Same thing in `visit_map` if possible.